pandas/io/formats/latex.py

```diff
@@ -106,13 +112,60 @@ def pad_empties(x):
             raise AssertionError('column_format must be str or unicode, '
                                  'not {typ}'.format(typ=type(column_format)))
 
+        use_table_env = False
         if not self.longtable:
+            if self.caption is None and self.label is None:
+                # then write output only in a tabular environment
+                pass
+            else:
+                # then write output in a nested table/tabular environment
+                use_table_env = True
+
+                if self.caption is None:
+                    caption_ = ''
+                else:
+                    caption_ = '\n\\caption{{{}}}'.format(self.caption)
+
+                if self.label is None:
+                    label_ = ''
+                    warnings.warn('no LaTeX label has been provided; '
+                                  'referencing with \\ref{} will not'
+                                  'be available')
+                else:
+                    label_ = '\n\\label{{{}}}'.format(self.label)
+
+                buf.write('\\begin{{table}}\n\\centering{}{}\n'.format(
+                    caption_,
+                    label_
+                ))
+
             buf.write('\\begin{{tabular}}{{{fmt}}}\n'
                       .format(fmt=column_format))
             buf.write('\\toprule\n')
         else:
             buf.write('\\begin{{longtable}}{{{fmt}}}\n'
                       .format(fmt=column_format))
+
+            if self.caption is None and self.label is None:
+                pass
+            else:
+                if self.caption is None:
+                    pass
+                else:
+                    buf.write('\\caption{{{}}}'.format(self.caption))
+
+                if self.label is None:
+                    warnings.warn('no LaTeX label has been provided; '
+                                  'referencing with \\ref{} will not'
+                                  'be available')
```

Comment:
I suspect we're going to need an explicit stack-level on this.  @jreback ?

Reply:
This warning has been removed after getting feedback from @tomneep below.